src/weakptr.c

```diff
@@ -37,6 +40,28 @@
 # include <gc/gc.h>
 #endif
 
+/****************************************************************************
+**
+*F
+*/
+
+#ifdef USE_GASMAN
+
+#define IS_WEAK_DEAD_BAG(bag) ( (((UInt)bag & (sizeof(Bag)-1)) == 0) && \
```

Comment:
This seems like a very low-level gasman thing to have in weakptr.h?

Reply:
Just to make sure we are on the same page: It's in `weakptr.c`, not in `weakptr.h` -- though before, it was indeed in a header file, namely in `gasman.h`.

tl;dr: yes, this is not ideal, but it's only meant to be temporary, so I hope we can let it slide for now...

Long version: Yes, this is rather low level GASMAN specific code -- but then so is a lot of other code in `weakptr.c`, you'll see lots of code specific to Boehm resp. GASMAN -- and with Julia GC, also code specific to *that*. IMHO this shows that the abstraction originally chosen to allow implementing weak pointer objects "without assuming too much about the GC internals" just does not work. (To be fair, I don't actually think that was a major concern when weak pointer objects were implemented, so it's of course natural that this non-goal was not achieved ;-).

So I think on the long run, we should consider refactoring this. But for now, it was simply convenient to hide the implementation specific parts somewhere, and here it seemed useful (for people reading the code, and trying to understand it) to have it in `weakptr.c`. But I could also put it into `gasman_intern.h` . The main reason I did not do this (yet!) is that there are also Boehm version of this macro; so then one might have to add a `boehm_gc.h` (which I think we'll do on the long run anyway). Anyway, this seemed to be more work to me, and perhaps a bit premature: I'd rather see how the Julia GC integration works out, and then see if we can come up with a better abstraction...